torch/_inductor/utils.py

```diff
@@ -2182,8 +2182,10 @@ def is_same_mkldnn_tensor(data: torch.Tensor, value: torch.Tensor):
 
 
 @dataclass_transform(frozen_default=True)
-def ir_dataclass(cls=None, /, *, frozen: bool = True):
-    def wrap(cls: _T) -> _T:
+def ir_dataclass(
```

Comment:
This entire function might be cleaner to write as just an if else using functools.partial